src/app/icd/ICDHandler.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines objects for a CHIP ICD handler which handles unsolicited checkin messages.
+ *
+ */
+
+#include "ICDHandler.h"
+
+#include <cinttypes>
+
+#include "access/RequestPath.h"
+#include "access/SubjectDescriptor.h"
+#include <app/util/endpoint-config-api.h>
+#include <lib/core/Global.h>
+#include <lib/core/TLVUtilities.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/FibonacciUtils.h>
+
+namespace chip {
+namespace app {
+namespace CheckIn {
+
+Global<CheckInMessageHandler> sCheckInMessageHandler;
+CheckInMessageHandler * CheckInMessageHandler::GetInstance()
```

Comment:
So we are going to have `app::CheckIn::CheckInMessageHandler` as the type name here, right?  Why the repeated `CheckIn`?